python/paddle/fluid/dygraph/dygraph_to_static/print_transformer.py

```diff
@@ -0,0 +1,94 @@
+#   Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import gast
+import astor
+
+from paddle.fluid.dygraph.dygraph_to_static.static_analysis import AstNodeWrapper, NodeVarType, StaticAnalysisVisitor
+
+
+class PrintTransformer(gast.NodeTransformer):
+    """
+    This class transform python print function to fluid.layers.Print.
+    """
+
+    def __init__(self, wrapper_root):
+        assert isinstance(
+            wrapper_root, AstNodeWrapper
+        ), "Input non-AstNodeWrapper node for the initialization of PrintTransformer."
+        self.wrapper_root = wrapper_root
+        self.root = wrapper_root.node
+
+        self.static_analysis_visitor = StaticAnalysisVisitor(self.root)
+        self.node_to_wrapper_map = self.static_analysis_visitor.get_node_to_wrapper_map(
+        )
+
+    def transform(self):
+        self.visit(self.root)
+
+    # NOTE: deal with print in PY3
+    def visit_Expr(self, node):
+        if isinstance(node.value, gast.Call):
+            node.value = self._visit_Call(node.value)
+        return node
+
+    def _visit_Call(self, node):
```

Comment:
Why not replace `_visit_Call` and `visit_Expr`  with `visit_Call` ? Maybe it's because of the custom print function like `a.print(...)` ?

Reply:
using `_visit_Call` and `visit_Expr` is because I think usually there is no such writing like `a = print(x)`, thanks for reminding